libsolidity/interface/CompilerStack.cpp

```diff
@@ -389,18 +389,27 @@ string const CompilerStack::lastContractName() const
 
 eth::AssemblyItems const* CompilerStack::assemblyItems(string const& _contractName) const
 {
+	if (m_stackState != CompilationSuccessful)
+		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Compilation was not successful."));
+
 	Contract const& currentContract = contract(_contractName);
```

Comment:
IIRC `contract` does this check already.

Reply:
still, wanted to play safe.